src/axis.ts

```diff
@@ -1,88 +1,92 @@
     g.append("g")
         .attr("transform", "translate(" + (svgBoundingClientRect.width - 100) + " ,0)")
-        .call(d3.axisRight(percentageAxis).tickPadding(2).ticks(10)
-        .tickFormat(function(d) {
-          return d + "%";
-        }));
+        .call(
+            d3
+                .axisRight(percentageAxis)
+                .tickPadding(2)
+                .ticks(10)
+                .tickFormat(function (d) {
+                    return d + "%";
+                })
+        );
 }
 
-const moduleCategoryAxis =(domain:any, rangeStart:number , rangeWidth:number)=>{
-
+const moduleCategoryAxis = (domain: any, rangeStart: number, rangeWidth: number) => {
     let categoryAxis = d3
-            .scaleBand()
-            .domain(domain)           
-            .range([rangeStart, rangeWidth - 100])
-            .paddingInner(0.13)
-            .paddingOuter(0.26)
+        .scaleBand()
+        .domain(domain)
+        .range([rangeStart, rangeWidth - 100])
+        .paddingInner(0.13)
+        .paddingOuter(0.26);
     return categoryAxis;
-
-}
-const moduleValueAxis = (domain:any,rangeHeight:number,ticks:number)=>{
+};
+const moduleValueAxis = (domain: any, rangeHeight: number, ticks: number) => {
     let valueAxis = d3
-            .scaleLinear()
-            .domain([0, domain]).nice(ticks)
-            .range([rangeHeight - 50, 100])
-        return valueAxis;
-}
-const modulePercentageAxis = (rangeHeight: number)=>{
-   let percentageAxis = d3
-    .scaleLinear()
-    .domain([0 , 100 ])
-    .range([rangeHeight - 50, 100]);
+        .scaleLinear()
+        .domain([0, domain])
+        .nice(ticks)
+        .range([rangeHeight - 50, 100]);
+    return valueAxis;
+};
+const modulePercentageAxis = (rangeHeight: number) => {
+    let percentageAxis = d3
+        .scaleLinear()
+        .domain([0, 100])
+        .range([rangeHeight - 50, 100]);
     return percentageAxis;
-}
-const moduleCategories = (pareto: Pareto)=>{
-    const paretoCategoryValues:string[] = [];
+};
+const moduleCategories = (pareto: Pareto) => {
+    const paretoCategoryValues: string[] = [];
     pareto.stackedBars.forEach((p) => {
         paretoCategoryValues.push(p.label);
     });
     return paretoCategoryValues;
-}
+};
 const moduleIndices = (pareto: Pareto)=>{
     const paretoCategoryIndices:number[] = [];
     pareto.stackedBars.forEach((p) => {
         paretoCategoryIndices.push(p.index);
     });
     return paretoCategoryIndices;
 }
-const moduleTicks = (pareto: Pareto)=>{
-    let ticks:number = 0;
-    if(pareto.maxValue <= 50){
-        ticks = 5;
-    }else if (pareto.maxValue >50 && pareto.maxValue <= 100){
-        ticks= 10;
-    }else {
-        ticks= 20;
-    }
+
+/**
+ * Calculates ticks based on height and font
+ * @param pareto
+ * @param height
+ * @param labelSize The size of the label font
+ * @returns Tick number
+ */
+const moduleTicks = (height: number, labelSize: number) => {
+ let    ticks = height / (labelSize * 2 + 6);
```

Comment:
A question before you merge @AstridBer, could you explain why you do labelSize * 2 + 6? In particular, why the number 6?

Reply:
I based this on how they handled the the ticks in the area chart. I played around with different numbers but thought this worked the best. The bigger the denominator is the less ticks you get. So if you want to render more ticks you would decrease and change 2 + 6 to a smaller number.